### PR TITLE
Add handler options

### PR DESCRIPTION
**Pull request by timostamm**
Update handler options to accept a type registry (along with other JSON or binary format options), or an array of protocols.

This also adds a helper for unimplemented services: Users can fill in a partial service implementation with methods that raise an `unimplemented` error.

### CODE REVIEW

File: packages/connect-node/src/connect-protocol.ts [hunk 1]
```diff
@@ -35,7 +35,7 @@ import {
 } from "@bufbuild/protobuf";
 import type { ImplHandler, ImplSpec, Protocol } from "./protocol.js";
 import { createServerMethodSerializers } from "./private/create-server-method-serializers.js";
-import type { HandlerContext } from "./handler.js";
+import type { HandlerContext } from "./implementation";
```
Comment (smaye81) [inline packages/connect-node/src/connect-protocol.ts:38]: `.js` extension?

File: packages/connect-node/src/handler.ts [hunk 2]
```diff
@@ -58,83 +52,148 @@ export type Handler = NodeHandler & {
...
+    handleProtocol(req, res).catch((reason) => {
+      // TODO need to handle rejections here, but it's unclear how exactly
+      // eslint-disable-next-line no-console
+      console.error("protocol handle failed", reason);
+    });
+  }
+  return Object.assign(handleAny, {
+    service,
+    method,
+    requestPath: createMethodUrl("https://example.com", service, method)
```
Comment (smaye81) [inline packages/connect-node/src/handler.ts:121]: Is this `example.com` url intended to be here?

**Review by smaye81 — APPROVED** [review]

File: packages/connect-node/src/connect-protocol.ts (hunk 1 above)
Comment (timostamm) [reply packages/connect-node/src/connect-protocol.ts]: Updated

File: packages/connect-node/src/handler.ts (hunk 2 above)
Comment (timostamm) [reply packages/connect-node/src/handler.ts:121]: The function requires a valid `baseUrl` argument, even though we only want the path of the resulting URL in the end.
I agree it's irritating to read, and I hope we find a better way to do this.

**Pull request merged**